default/python/AI/EspionageAI.py

```diff
@@ -112,6 +116,15 @@ def colony_detectable_by_empire(planet_id, species_name=None, empire=ALL_EMPIRES
 
     total_stealth = planet_stealth + sum([AIDependencies.STEALTH_STRENGTHS_BY_SPECIES_TAG.get(tag, 0)
                                           for tag in species_tags])
+
+    if verify_prediction:
```

Comment:
This whole verify_prediction section would need a number of additional adjustments and constraints on when it would consider logging a discrepancy, and even with those it's not really clear to me just when we could actually get a useful signal-to-noise ratio from this.

Although we are not currently actually using the ```future_stealth_bonus``` parameter, I plan to add support for AIs to pursue planetary stealth techs in the relatively near future, and then would use ```future_stealth_bonus``` to take such expected stealth bonus into account.   So before updating the planet_stealth (above) with ```future_stealth_bonus```, we could save ```prediction_assessment_value = planet_stealth``` and then do our tests here with respect to ```prediction_assessment_value``` rather than ```planet_stealth```

Also, anytime we have just Basic Visibility of an object, its stealth meter will be set to Meter::LARGE_VALUE ( 2^15 ), and would only be updated thereafter if we get Partial Visibility of it, so that established a large range of cases in which we would expect a mismatch between the predicted stealth and the meter reading and should not warn for.

But also if we had momentarily gotten Partial Visibility from an Effect like from a Derelict Scout (or on the turn that a stealthy species colonized the planet), and have not after that moment had any visibility of the object at all, then our current meter reading would still be that from the momentary glimpse of Partial Visibility (even if that is now stale and even if our current detection strength is below that level of stealth).